crates/extensions/c8y_mapper_ext/src/operations/log_upload.rs

```diff
@@ -207,21 +209,33 @@ impl CumulocityConverter {
             time: OffsetDateTime::now_utc(),
             text: response.log_type.clone(),
             extras: HashMap::new(),
-            device_id: target.external_id.as_ref().to_string(),
+            device_id: xid.to_string(),
         };
         let event_response_id = self.http_proxy.send_event(create_event).await?;
 
         let binary_upload_event_url = self
             .c8y_endpoint
             .get_url_for_event_binary_upload_unchecked(&event_response_id);
 
+        let file_path =
+            Utf8PathBuf::try_from(download_response.file_path).map_err(|e| e.into_io_error())?;
+
+        // The method must be POST, otherwise file name won't be supported.
+        // Mime must be text/*, otherwise c8y UI doesn't give a preview of the content.
         let upload_request = UploadRequest::new(
             self.auth_proxy
                 .proxy_url(binary_upload_event_url.clone())
                 .as_str(),
-            &Utf8PathBuf::try_from(download_response.file_path).map_err(|e| e.into_io_error())?,
+            &file_path,
         )
-        .with_content_type(ContentType::TextPlain);
+        .post()
+        .with_content_type(ContentType::FormData(
+            FormData::new(format!(
+                "{xid}_{filename}",
+                filename = file_path.file_name().unwrap_or("filename")
+            ))
+            .set_mime("text/plain".parse().unwrap()), // safe
```

Comment:
```suggestion
            .set_mime(mime::TEXT_PLAIN),
```

Reply:
Done in [62e808f](https://github.com/thin-edge/thin-edge.io/pull/2811/commits/62e808ff60b33c3c2f136d25c9dd8ff50d9b02be)